rskj-core/src/main/java/co/rsk/net/BlockSyncService.java

```diff
@@ -150,7 +150,23 @@ private void tryReleaseStore(long bestBlockNumber) {
         }
     }
 
+    public boolean isReadyToAcceptBlocks() {
+        if (blockchain.getSize() == 1) { // empty blockchain
+            return false;
```

Comment:
I'm not sure about this check. Intuitively, I would say that an empty blockchain should be ready to accept blocks, but I've seen dedicated tests suggesting the contrary:
- `co.rsk.net.NodeBlockProcessorTest#noSyncingWithEmptyBlockchain`
- `co.rsk.net.AsyncNodeBlockProcessorTest#noSyncingWithEmptyBlockchain`

Reply:
I have looked into this a bit. What I can say is that:
1. `isReadyToAcceptBlocks` refers specifically to a single type of message, `NewBlockHashMessage`, and probably the method should be renamed to reflect that.
2. Is possible, although surprising, that the intention behind the tests was to prevent blocks from receiving new block hashes they did not requests while they where not synced, similar to what is being checked now by checking the current known block number with zero. 

(2) should be confirmed. The code is a bit hard to follow. What I recommend is setting up a node and checking the logs. Writing down in detail the sync process wouldn't hurt either.

Reply:
Thanks Joaco, will take a look.